boto3/docs/action.py

```diff
@@ -119,7 +151,8 @@ def document_action(
     operation_model = service_model.operation_model(
         action_model.request.operation
     )
-    ignore_params = get_resource_ignore_params(action_model.request.params)
+    if ignore_params is None:
+        ignore_params = get_resource_ignore_params(action_model.request.params)
```

Comment:
Out of curiosity, because I'm not very familiar with this codebase: Why does the new argument override the previous method? Instead of giving the previous method precedence or merging the results of both?

Reply:
Well part of the purpose of this PR to is to override `get_resource_ignore_params` because it indicates to the doc generator to ignore too many parameters. For Metric.put_data`, it results in both `Namespace` and `MetricData[].MetricName` being ignored so no request/response examples end up in the docs. We can't merge them because we're trying to take away ignore params rather than add to them.

I can envision in the future where we'd have the opposite problem in that this function doesn't ignore enough parameters, but at the end of the day for these edge cases, I think its safest to just tell the generator exactly what we want to ignore rather than trying to be clever.